### PR TITLE
Added search API

### PR DESCRIPTION
**Pull request by keithhc2**
Signed-off-by: keithhc2 <keithhc2@users.noreply.github.com>

#299 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _opensearch/rest-api/search.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+```
+
+## URL Parameters
+
+All URL parameters are optional.
+
+Parameter | Type | Description
+:--- | :--- | :---
+allow_no_indexes | Boolean | Whether to ignore wildcards that don’t match any indexes. Default is true.
+allow_partial_search_results | Boolean | Whether to return partial results if the request runs into an error or timeouts. Default is true.
```
Comment (lizsnyder) [inline _opensearch/rest-api/search.md:44]: Should this be "or times out" instead of "or timeouts"?

File: _opensearch/rest-api/search.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+docvalue_fields | String | The fields that OpenSearch should return using their docvalue forms.
+expand_wildcards | String | Specifies the type of index that wildcard expressions can match. Supports comma-separated values. Valid values are all (match any index), open (match open, non-hidden indexes), closed (match closed, non-hidden indexes), hidden (match hidden indexes), and none (deny wildcard expressions). Default is open.
+explain | Boolean | Whether to return details about how OpenSearch computed the document's score. Default is false.
+from | Integer | The starting index to search from. Default is 0.
+ignore_throttled | Boolean | Whether to ignore concrete, expanded, or indexes with aliases if indexes are frozen. Default is true.
+ignore_unavailable | Boolean | Specifies whether to include missing or closed indexes in the response. Default is false.
+lenient | Boolean | Specifies whether OpenSearch should accept requests if queries have format errors (for example, querying a text field for an integer). Default is false.
+max_concurrent_shard_requests | Integer | The number of how many concurrent shard requests this request should execute on each node. Default is 5.
```
Comment (lizsnyder) [inline _opensearch/rest-api/search.md:58]: "The number of how many" is a little redundant

File: _opensearch/rest-api/search.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+preference | String | Specifies which shard or node OpenSearch should perform the count operation on.
+q | String | Lucene query string’s query.
+request_cache | Boolean | Specifies whether OpenSearch should use the request cache. Default is whether it’s enabled in the index’s settings.
+rest_total_hits_as_int | Boolean | Whether to return `hits.total` as an integer. Returns an object otherwise. Default is false.
+routing | String | Value used to route the update by query operation to a specific shard.
+scroll | Time | How long to keep the search context open.
+search_type | String | Whether OpenSearch should use global term and document frequencies calculating revelance scores. Valid choices are `query_then_fetch` and `dfs_query_then_fetch`. `query_then_fetch` scores documents using local term and document frequencies for the shard. It’s usually faster but less accurate. `dfs_query_then_fetch` scores documents using global term and document frequencies across all shards. It’s usually slower but more accurate. Default is `query_then_fetch`.
```
Comment (lizsnyder) [inline _opensearch/rest-api/search.md:66]: when calculating?

File: _opensearch/rest-api/search.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+_source_excludes | List | A comma-separated list of source fields to exclude from the response.
+_source_includes | List | A comma-separated list of source fields to include in the response.
+stats | String | Value to associate with the request for additional logging.
+stored_fields | Boolean | Whether the get operation should retrieve fields stored in the index. Default is false.
+suggest_field | String | Fields OpenSearch can use to look for similar terms.
+suggest_mode | String | The mode to use when searching. Available options are `always` (use suggestions based on the provided terms), `popular` (use suggestions that have more occurrences), and `missing` (use suggestions for terms not in the index).
+suggest_size | Integer | How many suggestions to return.
+suggest_text | String | The source that suggestions should be based off of.
+terminate_after | Integer | The maximum number of documents OpenSearch should process before terminating the request. Default is 0.
+timeout | Time | How long the operation should wait from a response from active shards. Default is `1m`.
```
Comment (lizsnyder) [inline _opensearch/rest-api/search.md:80]: wait for

**Review by lizsnyder — APPROVED** [review]

**Review by ashwinkumar12345 — APPROVED** [review]

**Pull request merged**